josh-core/src/history.rs

```diff
@@ -196,49 +196,70 @@ pub struct RewriteData<'a> {
     pub message: Option<String>,
 }
 
-// takes everything from base except it's tree and replaces it with the tree
+// takes everything from base except its tree and replaces it with the tree
 // given
 pub fn rewrite_commit(
     repo: &git2::Repository,
-    base: &git2::Commit,
+    base: &git2::Commit, // original commit that we are modifying
     parents: &[&git2::Commit],
     rewrite_data: RewriteData,
     unsign: bool,
 ) -> JoshResult<git2::Oid> {
-    let message = rewrite_data
-        .message
-        .unwrap_or(base.message_raw().unwrap_or("no message").to_string());
-    let tree = &rewrite_data.tree;
-
-    let a = base.author();
-    let new_a = if let Some((author, email)) = rewrite_data.author {
-        git2::Signature::new(&author, &email, &a.when())?
-    } else {
-        a
-    };
+    use bstr::ByteSlice;
```

Comment:
I'd prefer to not have `use` but instead fully qualify the identifiers in place.

Reply:
Hmm... this is a trait not a module, though...

In any case, it looks like this is somewhat vestigial from an earlier approach. The only place this is needed can be replaced with a deref. This is done in 053d5e35ca7d48173b32723064eac3bef53fc48a

I'm interpreting this feedback to also apply to `use gix_object::{CommitRef, WriteTo};` but have broken it out as a separate commit 31ab08d43d837e9ac09b96c96379b33e5031ce02 in case not.